### PR TITLE
ARROW-15721: [Website] Add Flight/Flight SQL to subprojects

### PR DESCRIPTION
**Pull request by lidavidm**

### CODE REVIEW

**Comment by lidavidm** [comment]
Depends on https://github.com/apache/arrow/pull/12573 and links to the dev docs for Flight SQL.

File: _includes/header.html [hunk 1]
```diff
@@ -55,6 +55,8 @@
              Subprojects
           </a>
           <div class="dropdown-menu" aria-labelledby="navbarDropdownSubprojects">
+            <a class="dropdown-item" href="{{ site.baseurl }}/docs/format/Flight.html">Arrow Flight</a>
+            <a class="dropdown-item" href="{{ site.baseurl }}/docs/dev/format/FlightSql.html">Arrow Flight SQL</a>
```
Comment (kou) [inline _includes/header.html:59]: We need to remove `dev/` here after we release 8.0.0, right?

**Review by kou — APPROVED** [review]
+1

File: _includes/header.html (hunk 1 above)
Comment (lidavidm) [reply _includes/header.html:59]: Yes, good point. I filed ARROW-15937 as a reminder.

**Pull request closed without merging**